# Describing Data

## Brady Bunch Data Summary

Source: Questions from Thinkful Curriculum

This file written by: Dipti Muni
Date: 2-8-2018

In [1]:
#Question 1-5
import pandas as pd # used for computation
import numpy as np # used for mean
import statistics 
from statistics import mode, StatisticsError

### Question One

__Greg was 14, Marcia was 12, Peter was 11, Jan was 10, Bobby was 8, and Cindy was 6 when they started playing the Brady kids on The Brady Bunch. Cousin Oliver was 8 years old when he joined the show. What are the mean, median, and mode of the kids' ages when they first appeared on the show? What are the variance, standard deviation, and standard error?__


#### Central Tendency

In [2]:
# Make a blank data frame.
df = pd.DataFrame()

# Populate it with data.
df['age'] = [14, 12, 11, 10, 8, 6, 8]

In [3]:
df.describe()

age
count   7.000000
mean    9.857143
std     2.734262
min     6.000000
25%     8.000000
50%    10.000000
75%    11.500000
max    14.000000

In [4]:
mean_age = np.mean(df['age'])
print("The mean is " + str(round(mean_age, 2)) + ".")

The mean is 9.86.


In [5]:
median_age = np.median(df['age'])
print("The median is " + str(round(median_age, 2)) + ".")

The median is 10.0.


In [6]:
try:
    mode_age = statistics.mode(df['age'])
    print("The mode is {}.".format(mode_age))
except StatisticsError:
    print("There is no unique mode!")

The mode is 8.


#### Variance

In [7]:
variance_age = df['age'].var()
print("The variance is " + str(round(variance_age, 2)) + ".")

The variance is 7.48.


In [8]:
sd_age = np.std(df['age'], ddof=1)
print("The standard deviation is " + str(round(sd_age, 2)) + ".")

The standard deviation is 2.73.


In [9]:
se_age = np.std(df['age'] ,ddof=1) / np.sqrt(len(df['age']))
print("The standard error is " + str(round(se_age, 2)) + ".")

The standard error is 1.03.


### Question Two

__Using these estimates, if you had to choose only one estimate of central tendency and one estimate of variance to describe the data, which would you pick and why?__

For estimate of central tendency, there is not much difference in the mean and the median. Hence, I would choose either because age of 10 (approximately speaking) seems to be the average in the Brady Bunch group. 

Next, variance of the age variable should describe how much values differ from the central tendency (which is about 10). Therefore, I would say standard deviation estimate of 2.73 closely describes the average difference from central tendency. For example, for the sake of simplicity, let's say the mean is 10, so if we were to then subtract mean from all of the ages you get [4, 2, 1, 0, 2, 4, 2] in absolute value and divide the total by 7 you will get 2.14 average difference from 10. 

### Question Three

__Next, Cindy has a birthday. Update your estimates- what changed, and what didn't?__

In [10]:
df['age'] = [14, 12, 11, 10, 8, 7, 8]
def calculation(data):
#Mean
    mean_age = np.mean(df['age'])
    print("The mean is " + str(round(mean_age, 2)) + ".")

#Median
    median_age = np.median(df['age'])
    print("The median is " + str(round(median_age, 2)) + ".")

#Mode
    try:
        mode_age = statistics.mode(df['age'])
        print("The mode is {}.".format(mode_age))
    except StatisticsError:
        print("There is no unique mode!")
    
#Variance
    variance_age = df['age'].var()
    print("The variance is " + str(round(variance_age, 2)) + ".")

#Standard Deviation
    sd_age = np.std(df['age'], ddof=1)
    print("The standard deviation is " + str(round(sd_age, 2)) + ".")

#Standard Error
    se_age = np.std(df['age'] ,ddof=1) / np.sqrt(len(df['age']))
    print("The standard error is " + str(round(se_age, 2)) + ".")
    
calculation(df['age'])


The mean is 10.0.
The median is 10.0.
The mode is 8.
The variance is 6.33.
The standard deviation is 2.52.
The standard error is 0.95.


Seeing the central tendency and variance calculations above, we can see that the median and mode are the only two that has not changed in their answer. The mean increased by .14. The variance decreased by 1.15. The standard deviation decreased by .21. The standard error decreased by .08. 

What does this mean when Cindy got one year older? She is getting closer to the mean average and therefore the variance between different values has desreased a bit as well. If everyone below median gets one year older without change in everyone above median, we get increase mean and gaps in variances gets smaller. 

### Question Four

__Nobody likes Cousin Oliver. Maybe the network should have used an even younger actor. Replace Cousin Oliver with 1-year-old Jessica, then recalculate again. Does this change your choice of central tendency or variance estimation methods?__

In [11]:
df['age'] = [14, 12, 11, 10, 8, 7, 1]
calculation(df['age'])

The mean is 9.0.
The median is 10.0.
There is no unique mode!
The variance is 18.0.
The standard deviation is 4.24.
The standard error is 1.6.


Yes, for central tendency I would stick with median only. Because Jessica is the only one that is very young, it askews the data. As for my choice for variance estimation method. The variance increases signicantly by 11.67 and is not reliable when you have outlier. The standard deviation increases by 1.72 and the standard error increases by .65. I would stick with standard error to help with estimating the accuracy of my data. 

### Question Five

__On the 50th anniversary of The Brady Bunch, four different magazines asked their readers whether they were fans of the show. The answers were: TV Guide 20% fans Entertainment Weekly 23% fans Pop Culture Today 17% fans SciPhi Phanatic 5% fans. Based on these numbers, what percentage of adult Americans would you estimate were Brady Bunch fans on the 50th anniversary of the show?__

In [12]:
#We will leave SciPhi Phanatic fans out because it is a very specific kind of fans and does not represent the general population. Therefore, it will cause misrepresentation of the estimate. 
adult_percent =((20 + 23 + 17) / 300) * 100
print("On the 50th anniversary, the estimate number of " + str(adult_percent) + "% adult Americans were Brady Bunch Fans.")

On the 50th anniversary, the estimate number of 20.0% adult Americans were Brady Bunch Fans.
